### DISCUSSION TITLE
MSB4062 error of System.Private.Xml build

### DISCUSSION THREAD

**Discussion by Maximys** [dotnet/runtime#92126] [Q&A]
Hello guys.
Could anybody from .NET Team explain me what's wrong with building [System.Private.Xml](https://github.com/dotnet/runtime/tree/main/src/libraries/System.Private.Xml) by VisualStudio 17.8.0 Preview 2.0? After [faf883db18d04fc15226edcbce25035e1e548721](https://github.com/dotnet/runtime/commit/faf883db18d04fc15226edcbce25035e1e548721) commit I could not successfully build this solution by VisualStudio anymore. I have next error message:

> "MSB4062 The "CombineLinkerXmlFiles" task could not be loaded from the assembly C:\runtime\artifacts\bin\ILLink.Tasks\Debug\net8.0/ILLink.Tasks.dll. Could not load file or assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask."

**Comment by danmoseley** [comment]
@ViktorHofer

**Comment by ViktorHofer** [comment]
The solution files in dotnet/runtime libs need to be updated as the linker is now a live dependency. Will submit a PR. cc @sbomer

EDIT:
https://github.com/dotnet/runtime/pull/92209

**Reply by Maximys** [reply]
@ViktorHofer , I still have the same error. May be there are some other references, which should be fixed?

**Reply by DanielLyons178** [reply]
I've been unable to build the tests in [Microsoft.Extensions.Configuration.Binder](https://github.com/dotnet/runtime/tree/main/src/libraries/Microsoft.Extensions.Configuration.Binder) too with the same error.

**Reply by ViktorHofer** [reply]
Will take a look when I'm back from holiday

**Reply by Maximys** [reply]
@ViktorHofer , do you have any progress with current problem?

**Reply by ViktorHofer** [reply]
thanks for the reminder. taking a look now.

**Reply by ViktorHofer** [reply]
Trying out a fix. See https://github.com/dotnet/runtime/pull/92928

**Comment by karakasa** [comment]
I got the same issue today with VS 17.8.0 P2 in `System.IO.FIle.AccessControl`. Also it seems `ILLink.Tasks.dll` is locked by MSBuild initated by the building process itself. `System.Runtime.dll` is there but the building process cannot find it.

I run into MSB3021/3027/4062/4018. #92478

**Comment by ViktorHofer** [comment]
Should now be finally fixed. Thanks for the reports everyone.
